Sources/SwiftProtobuf/Message+AsyncMessageSequence.swift

```diff
@@ -0,0 +1,232 @@
+  ///           `BinaryDecodingError.truncated` if the sequence ends before fully decoding a
+  ///           message or a delimiter,
+  ///           `BinaryDecodingError.malformedProtobuf`if a delimiter could not be read and
+  ///           `BinaryDecodingError.tooLarge` if a size delimiter of 2GB or greater is found.
+  ///
+public static func asyncSequence<Base: AsyncSequence> (
+    baseSequence: Base,
+    extensions: ExtensionMap? = nil,
+    partial: Bool = false,
+    options: BinaryDecodingOptions = BinaryDecodingOptions()
+  ) -> AsyncMessageSequence<Base, Self> {
+    AsyncMessageSequence<Base, Self>(
+      baseSequence: baseSequence,
+      extensions: extensions,
+      partial: partial,
+      options: options
+    )
+  }
+}
+
+@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+extension AsyncSequence {
+  
+  /// Creates an asynchronous sequence of size-delimited messages from this sequence of bytes.
+  /// Delimited format allows a single file or stream to contain multiple messages. A delimited message
+  /// is a varint encoding the message size followed by a message of exactly that size.
+  ///
+  /// - Parameters:
+  ///   - messageType: The  type of message to read.
+  ///   - extensions: An `ExtensionMap` used to look up and decode any extensions in
+  ///    messages encoded by this sequence, or in messages nested within these messages.
+  ///   - partial: If `false` (the default),  after decoding a message, `Message.isInitialized`
+  ///     will be checked to ensure all fields are present. If any are missing,
+  ///     `BinaryEncodingError.missingRequiredFields` will be thrown.
+  ///   - options: The BinaryDecodingOptions to use.
+  /// - Returns: An asynchronous sequence of messages read from the `AsyncSequence` of bytes.
+  /// - Throws: `BinaryDecodingError` if decoding fails, throws
+  ///           `BinaryDelimited.Error` for some reading errors,
+  ///           `BinaryDecodingError.truncated` if the stream ends before fully decoding a
+  ///           message or a delimiter,
+  ///           `BinaryDecodingError.malformedProtobuf`if a delimiter could not be read and
+  ///           `BinaryDecodingError.tooLarge` if a size delimiter of 2GB or greater is found.
+  public func decodedBinaryDelimitedMessages<M: Message>(
+    messageType: M.Type,
+    extensions: ExtensionMap? = nil,
+    partial: Bool = false,
+    options: BinaryDecodingOptions = BinaryDecodingOptions()
+  ) -> AsyncMessageSequence<Self, M> {
+    AsyncMessageSequence<Self, M>(
+      baseSequence: self,
+      extensions: extensions,
+      partial: partial,
+      options: options
+    )
+  }
+}
+
+@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+/// An asynchronous sequence of messages decoded from an asynchronous sequence of bytes.
+public struct AsyncMessageSequence<Base: AsyncSequence, M: Message>:
+  AsyncSequence, Sendable
+where Base: Sendable, Base.Element == UInt8 {
```

Comment:
```suggestion
public struct AsyncMessageSequence<
    Base: AsyncSequence,
    M: Message
>: AsyncSequence Base.Element == UInt8 {
```

Reply:
Can we please conditionalise the extension on Sendable at the bottom with a

```
extension AsyncMessageSequence: Sendable where Base: Sendable
```

Reply:
Done.